modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/SnapshotDownloadStorage.scala

```diff
@@ -37,6 +36,34 @@ object SnapshotDownloadStorage {
 
       def isPersisted(hash: Hash): F[Boolean] = persistedStorage.exists(hash)
 
+      def hasSnapshotInfo(ordinal: SnapshotOrdinal): F[Boolean] =
+        snapshotInfoStorage.exists(ordinal)
+
+      def hasCorrectSnapshotInfo(ordinal: SnapshotOrdinal, proof: GlobalSnapshotStateProof): F[Boolean] =
+        snapshotInfoStorage.read(ordinal).flatMap {
+          case Some(info) => info.stateProof.map(_ === proof)
+          case _          => false.pure[F]
+        }
+
+      def getHighestSnapshotInfo(lte: SnapshotOrdinal): F[Option[SnapshotOrdinal]] =
+        snapshotInfoStorage.listStoredOrdinals.flatMap(_.filter(_ <= lte).compile.toList).map(_.maximumOption)
```

Comment:
Maybe break into multiple lines to be more readable?

Reply:
code formatting is done automatically by scalafix and scalafmt